sky/skylet/events.py

```diff
@@ -116,7 +119,8 @@ def _run(self):
             logger.debug('autostop_config not set. Skipped.')
             return
 
-        if job_lib.is_cluster_idle():
+        if job_lib.is_cluster_idle() and managed_job_state.get_num_alive_jobs(
+        ) == 0:
             idle_minutes = (time.time() -
                             autostop_lib.get_last_active_time()) // 60
```

Comment:
The `last_active_time` will be inaccurate now, and the controller will be stopped immediately after the last managed job finishes? Should we update the last_active_time` whenever a controller exits?

Reply:
This works the same way as for normal jobs. If this `if` is false, it will set the last active time to the current time. If true, it checks if the time since the last active time is more than the autostop time. It is inaccurate up to the event interval (60s).
There is no change from how this works before. The only reason we need this change is that is_cluster_idle will no longer be false if there are managed jobs running.

Reply:
Ahh, good point! This function also updates the last active time. Makes sense! Thanks for the explanation.